### ISSUE TITLE
Poor draw_networkx_labels result

### ISSUE THREAD

**Issue by ramdhan1989** [networkx/networkx#6877]
Hi, anyone can help me how to fix draw_network_labels plot result?  graphnx is the graph object.
this is graphnx NodeView(('v_(1, 6)', 'v_(1, 7)', 'v_(2, 5)', 'v_(2, 7)', 'v_(2, 4)', 'v_(2, 3)', 'v_(3, 5)', 'v_(1, 2)', 'v_(1, 8)', 'v_(1, 1)', 'v_(2, 0)', 'v_(0, 2)', 'v_(1, 9)', 'v_(1, 10)', 'v_(1, 11)', 'v_(1, 12)', 'v_(0, 13)', 'v_(1, 13)', 'v_(2, 13)', 'v_(3, 8)'))

I would like draw according to the coordinate position as follow:
node_coor = {'v_(1, 6)': [1, 6], 'v_(2, 4)': [2, 4], 'v_(2, 3)': [2, 3], 'v_(1, 7)': [1, 7], 'v_(1, 1)': [1, 1], 'v_(1, 8)': [1, 8], 'v_(1, 10)': [1, 10], 'v_(2, 0)': [2, 0], 'v_(1, 11)': [1, 11],'v_(0, 2)': [0, 2],'v_(1, 9)': [1, 9],'v_(2, 5)': [2, 5], 'v_(1, 12)': [1, 12],'v_(2, 7)': [2, 7],'v_(0, 13)': [0, 13],'v_(1, 13)': [1, 13],
 'v_(3, 5)': [3, 5],'v_(1, 2)': [1, 2], 'v_(2, 13)': [2, 13],'v_(3, 8)': [3, 8]}

executing this code : edge_labels=nx.draw_networkx_labels(graphnx,pos=node_coor) resulting plot as follow :  
![image](https://github.com/networkx/networkx/assets/41245856/7e142165-5155-4e34-9430-cfe6b0acc14e)

This also happened when executing : nx.draw_networkx_edge_labels(graphnx,pos=node_coor) 

Thank you

**Comment by dschult** [comment]
I can see in the upper right corner part of the text of the labels. So it is drawing, but apparently off the screen.
I suspect you need to change the limits on the axes of the plot. Are you also plotting the nodes? By that I mean, has matplotlib received any commands to create the axis limits?  I'm not sure drawing edge labels by themselves will do it.

**Comment by rossbar** [comment]
> Are you also plotting the nodes? By that I mean, has matplotlib received any commands to create the axis limits? I'm not sure drawing edge labels by themselves will do it.

Yeah I suspect that the `_labels` drawing functions implicitly assume that at least one of the functions for drawing nodes/edges/entire graphs has already been called to properly scale the axes.

This could likely be "fixed" by adding axis scaling to the label-drawing functions (i.e. something like `ax.set_xlim(pos[:, 0].min(), pos[:, 0].max())`, but this will likely give slightly different scaling than e.g. `ax.scatter` when draw_nodes is called. In other words, "fixing" this may slightly existing visualizations and introduce a dependence on calling order (i.e.

```python
>>> nx.draw_networkx_nodes(G, pos)
>>> nx.draw_networkx_labels(G, pos)
```

Would result in a different axis scaling than

```python
>>> nx.draw_networkx_labels(G, pos)
>>> nx.draw_networkx_nodes(G, pos)
```
)

Given these potential downsides, I'm inclined to simply document this quirk.

**Comment by akshitasure12** [comment]
From the discussion, I understand that to ensure proper visualization, we should explicitly document the need to set the axis scaling manually. The order in which we call the drawing functions `nx.draw_networkx_nodes` and `nx.draw_networkx_labels` can impact the axis scaling. This is because Matplotlib adjusts the axis limits dynamically, and this behavior may lead to slight inconsistencies between visualizations when the drawing order changes.

To fix this, we can explicitly set the axis limits using the node positions. However, this manual adjustment may slightly alter the scaling and potentially cause minor differences between visualizations, depending on the calling order of drawing functions.

Can I add these changes to the documentation and raise a PR?
